pkgs/os-specific/linux/cifs-utils/default.nix

```diff
@@ -14,6 +14,8 @@ stdenv.mkDerivation rec {
 
   buildInputs = [ kerberos keyutils pam talloc ];
 
+  patches = lib.optionals (stdenv.hostPlatform != stdenv.buildPlatform) [ ./no-AC_FUNC_MALLOC.patch ];
```

Comment:
To avoid the patch, could you try something like this:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/filesystems/nilfs-utils/default.nix#L40-L43

Reply:
It works, thank you!